code/pages/ExtensibleSearchPage.php

```diff
@@ -719,11 +748,76 @@ public function getSearchResults($data = null, $form = null) {
 
 		$startTime = microtime(true);
 
+		if($results = $this->getSearchEngineResults($data, $form)) {
```

Comment:
To address things properly, the fulltext implementation should be probably implemented as its own SearchEngine type rather than as fallback-type functionality, and the logic in ExtensibleSearchPage[_Controller] should be doing things like
```
$engine = $this->injector->get($this->SearchEngine);
$results = $engine->getSearchResults();
```
rather than interrogating its bound set of extensions looking for a name match and handing control off that way. It makes the interface much clearer then that the "Solr" implementation should be considered an _alternative_ to the "FullText" implementation. 

Generally speaking, using a 'fallback' if the specified search engine finds nothing doesn't feel correct to me, as there's possibly very good reasons something shouldn't be found in a query executed against your chosen search engine that may be matched in the 'fallback' fulltext search. If you really wanted that fallback type functionality, you could then create a 'CascadingSearch' SearchEngine type that first did searches against the Solr SearchEngine type, then the FullText SearchEngine type.